src/common/io-config/src/s3.rs

```diff
@@ -1,16 +1,20 @@
-use std::fmt::Display;
-use std::fmt::Formatter;
-
+#[cfg(feature = "python")]
+use crate::s3_provider::S3CredentialsProvider;
 use serde::Deserialize;
 use serde::Serialize;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::hash::Hash;
 
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
 pub struct S3Config {
     pub region_name: Option<String>,
     pub endpoint_url: Option<String>,
     pub key_id: Option<String>,
     pub session_token: Option<String>,
     pub access_key: Option<String>,
+    #[cfg(feature = "python")]
```

Comment:
You dont have to gate this behind python! we should instead feature-gate inside `S3CredentialsProvider` instead.